call.go

```diff
@@ -130,3 +133,170 @@ func WithRoutingKey(rk string) CallOption {
 func WithRoutingDelegate(rd string) CallOption {
 	return CallOption{func(o *OutboundCall) { o.routingDelegate = &rd }}
 }
+
+// InboundCall is an incoming call. It holds information about the inbound
+// call and its response.
+//
+// Encoding authors may use InboundCall to provide information about the
+// incoming request on the Context and receive response headers through
+// WriteResponseHeader.
+type InboundCall struct {
+	resHeaders []keyValuePair
+	req        *transport.Request
+}
+
+type inboundCallKey struct{} // context key for *InboundCall
+
+// NewInboundCall builds a new InboundCall with the given context.
+//
+// A request context is returned and must be used in place of the original.
+func NewInboundCall(ctx context.Context) (context.Context, *InboundCall) {
+	call := &InboundCall{}
+	return context.WithValue(ctx, inboundCallKey{}, call), call
```

Comment:
Using an unexported key has the effect of no other package being able to access this key?